codegen/index.js

```diff
@@ -76,7 +79,8 @@ const propType = (e, p) => {
 const printRow = (e, elProps) =>
   elProps.length > 0
     ? `
-  ( ${elProps.map(p => `${p} :: ${propType(e, p)}`).join("\n  , ")}
+  ( _data :: Object String
```

Comment:
`Object String` will not work with the intended usage of `_data: { foo: "bar" }`
To convert the record to `Object String`, you need `fromHomogeneous`.

Reply:
Yep, I fixed that 👍